src/main/java/nwoolcan/model/brewery/warehouse/article/ArticleManager.java

```diff
@@ -0,0 +1,137 @@
+package nwoolcan.model.brewery.warehouse.article;
+
+import nwoolcan.model.utils.UnitOfMeasure;
+import nwoolcan.utils.Result;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Manager for {@link Article} objects. It is used by tests and by the
+ * {@link nwoolcan.model.brewery.warehouse.Warehouse} in order to create {@link Article}, check the
+ * id of the {@link Article}, avoid repetitions and set name of the {@link Article}.
+ */
+public final class ArticleManager {
```

Comment:
Just an observation, you can make it lazy and thread safe without syncronized as explained [here](https://iol.unibo.it/pluginfile.php/302476/mod_folder/content/0/21-Patterns.pdf?forcedownload=1) silde 30

Reply:
I saw that method and I liked it, but would it free me of `synchronize` in other methods? For example adding article or modifying them.. Just because JUnit runs everything in parallel and it could get something wrong..

Reply:
i didn't remember what viroli said and why it is thread safe, but the slide reports that as thread-safe.
maybe some kind of magic with nested static classes?

Reply:
I get that it is completely thread safe because it is instantiated only the first time you call that class. it is just a trick. I am thinking about the thread safeness of the other methods, not the constructor..

Reply:
Yeah right ofc. I should fix mine too then